### DISCUSSION TITLE
Having trouble using Turbo with NPM

### DISCUSSION THREAD

**Discussion by H3RSKO** [vercel/turborepo#325] [Help] [answered]
So I have a simple monorepo with a `/client` folder containing a vite react project, and a `/server` folder containing a simple express server.

I'm using NPM, and even after going through the documentation several times I am unable to get turborepo to work. 

I have installed turbo in the root folder of my project, and the root package.json looks like this:

```
{
  "name": "test-app",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "turbo": "^1.0.9"
  },
  "private": true,
  "workspaces": [
    "client/*",
    "server/*"
  ],
  "turbo": {
    "npmClient": "npm",
    "pipeline": {
      "build": {
        "dependsOn": [
          "^build"
        ]
      },
      "lint": {
        "outputs": []
      },
      "dev": {
        "cache": false
      }
    }
  }
}

``` 
I have two issues:

1. The documentation says to run `yarn turbo run build`, but i'm using NPM, so what is the correct command to run? The documentation does not say anything about adding scripts to my root package.json, so running a variation of `npm run turbo run build` errors with "missing script 'turbo'". Running `npm turbo run build` errors with "unknown command". Running `turbo run build` will have the same error. 
2. If i do run `yarn turbo run build` as listed in the documentation nothing is actually built. From what I understand this should essentially be running `yarn run build` in my `./client` folder, but nothing is being run. 
<img width="267" alt="Screen Shot 2021-12-16 at 11 55 53 AM" src="https://user-images.githubusercontent.com/34112131/146416042-dbafe68f-74e0-4d34-87e7-e104d020dbe0.png">

I'm sure i'm making a simple mistake, but any help would be appreciated. Thanks!

**Comment by vittolewerissa** [comment, marked as answer]
AFAIK,

You could either run using `npx` such like `npx turbo run build` or just setup a script, ex:
```js
{
    "build": "turbo run build",
    "dev": "turbo run dev --parallel",
    "lint": "turbo run lint"
}
```

and run it on the project root.

**Reply by H3RSKO** [reply]
This was it. Thanks!
